spec/lib/whitehall_importer/migrate_assets_spec.rb

```diff
@@ -0,0 +1,143 @@
+      whitehall_import = build(:whitehall_migration_document_import, assets: [asset])
+      expect(asset).to receive(:update!).with(state: "processing").once.ordered
+      expect(asset).to receive(:update!).with(hash_including(:state)).once.ordered
+      described_class.call(whitehall_import)
+    end
+
+    it "should skip migrating any assets that have already been processed" do
+      asset = create(:whitehall_migration_asset_import, state: "removed")
+      whitehall_import = build(:whitehall_migration_document_import, assets: [asset])
+      expect(asset).not_to receive(:update!)
+      described_class.call(whitehall_import)
+    end
+
+    it "should log GdsApi errors and put into an aborted state" do
+      whitehall_import = build(:whitehall_migration_document_import, assets: [asset])
+      allow(asset).to receive(:whitehall_asset_id).and_raise(GdsApi::BaseError.new("Some problem with GDS API"))
+      described_class.call(whitehall_import)
+      expect(asset.state).to eq("migration_failed")
+      expect(asset.error_message).to include("Some problem with GDS API")
+    end
+
+    it "should log standard errors and put into an aborted state" do
+      whitehall_import = build(:whitehall_migration_document_import, assets: [asset])
+      allow(asset).to receive(:whitehall_asset_id).and_raise(StandardError.new("Some error"))
+      described_class.call(whitehall_import)
+      expect(asset.state).to eq("migration_failed")
+      expect(asset.error_message).to include("Some error")
+    end
+
+    it "should delete draft assets" do
+      asset = create(:whitehall_migration_asset_import)
+      delete_asset_request = stub_asset_manager_delete_asset(asset_id)
+      whitehall_import = build(:whitehall_migration_document_import, assets: [asset])
+
+      described_class.call(whitehall_import)
+      expect(delete_asset_request).to have_been_requested
+      expect(asset.state).to eq("removed")
+    end
+
+    it "should delete draft asset variants" do
+      asset = create(:whitehall_migration_asset_import, :for_image, variant: "s300")
+      delete_asset_request = stub_asset_manager_delete_asset(asset_id)
+      whitehall_import = build(:whitehall_migration_document_import, assets: [asset])
+
+      described_class.call(whitehall_import)
+      expect(delete_asset_request).to have_been_requested
+      expect(asset.state).to eq("removed")
+    end
+
+    it "should redirect live attachments to their content publisher equivalents" do
+      asset = create(:whitehall_migration_asset_import,
+                     :for_file_attachment,
+                     original_asset_url: original_asset_url)
+      whitehall_import = build(:whitehall_migration_document_import, assets: [asset])
+      new_file_url = "https://asset-manager.gov.uk/NEW/847151/foo.pdf"
+      allow(asset.file_attachment_revision.asset).to receive(:live?).and_return(true)
+      allow(asset.file_attachment_revision.asset).to receive(:file_url).and_return(new_file_url)
```

Comment:
Can we create the file_attachment_revision with these attributes rather than needing to mock them?

Similar point on the tests below.